adr/0001-recipe-marketplace-csv-gradle-tasks.md

```diff
@@ -0,0 +1,178 @@
+# 1. Recipe Marketplace CSV Gradle Tasks
+
+Date: 2025-10-26
+
+## Status
+
+Accepted
+
+## Context
+
+Recipe library projects need a standardized way to generate and maintain `recipes.csv` files that describe their recipe marketplace. The Recipe Marketplace CSV format (defined in [rewrite ADR-006](https://github.com/openrewrite/rewrite/blob/main/doc/adr/0006-recipe-marketplace-csv-format.md)) provides a human-readable format for documenting:
+
+- Recipe names, display names, and descriptions
+- Recipe categorization in hierarchies
+- Recipe options with their metadata
+- Optional bundle information (ecosystem, package name, version)
+
+However, maintaining these CSV files manually is error-prone and can easily become out of sync with the actual recipes in the codebase. We need Gradle tasks that:
+
+1. **Automatically generate** `recipes.csv` from recipe JARs
+2. **Merge** generated data with existing CSV customizations
+3. **Validate** CSV content for formatting rules (display names, descriptions)
+4. **Validate** CSV completeness to ensure synchronization between CSV and JAR
+
+These tasks should be available to any project using the `RewriteRecipeLibraryPlugin` or `RewriteRecipeLibraryBasePlugin`.
```

Comment:
My first thought at this line is what about language libraries? For instance rewrite-toml, uses:
```
plugins {
    id("org.openrewrite.build.language-library")
}
```

Which then includes
https://github.com/openrewrite/rewrite-build-gradle-plugin/blob/42874c5215d1188465841f2a6a929ae7625cfc53/build.gradle.kts#L49-L55

But that does not provide these new tasks, meaning some recipes would not see any CSV created.